api/tests/test_token_checker.py

```diff
@@ -0,0 +1,397 @@
+        payload_list.append(self._standard_token())
+
+        for payload in payload_list:
+            token = self._create_token(payload, FORGED_PRIVATE_KEY)
+            with self.settings(IAM_HOSTNAME_LIST=['iam-test.idc.eu']):
+                self.assertFalse(
+                    self._token_checker._verify_token(token, payload['iss']),
+                    "Payload %s should not be accepted!" % payload
+                )
+
+                self.assertEqual(
+                    self._token_checker.valid_token_to_id(token), None,
+                    "Token with payload %s should not be accepted!" % payload
+                )
+
+    def test_is_token_issuer_trusted(self):
+        """
+        Check an untrusted 'issuer' (or missing 'issuer') is detected.
+
+        Both by:
+         - _is_token_issuer_trusted
+         - valid_token_to_id
+
+        The first method checks wether the issuer is
+        in settings.IAM_HOSTNAME_LIST
+        The second method detemines wether the token is invalid
+        """
+        payload_list = []
+
+        # Test we reject a payload without 'iss' field
+        # as we cannot tell where it came from (so can't verify it)
+        payload_list.append({
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'iat': int(time.time()) - 2000000,
+            'sub': CLIENT_ID,
+            'exp': int(time.time()) + 200000}
+        )
+
+        # Test we reject a payload with a malicious 'iss' field
+        # as we do not want to attempt to verify it
+        payload_list.append({
+            'iss': 'https://malicious-iam.idc.biz/',
+            'jti': '098cb343-c45e-490d-8aa0-ce1873cdc5f8',
+            'iat': int(time.time()) - 2000000,
+            'sub': CLIENT_ID,
+            'exp': int(time.time()) + 200000}
+        )
+
+        for payload in payload_list:
+            token = self._create_token(payload, PRIVATE_KEY)
+
+            with self.settings(IAM_HOSTNAME_LIST=['iam-test.idc.eu']):
+                self.assertFalse(
+                    self._token_checker._is_token_issuer_trusted(payload),
+                    "Payload %s should not be accepted!" % payload
+                )
+
+                self.assertEqual(
+                    self._token_checker.valid_token_to_id(token), None,
+                    "Token with payload %s should not be accepted!" % payload
+                )
+
+    def test_is_token_json_temporally_valid(self):
+        """
+        Check that temporally invalid payload/token is detected.
+
+        Both by:
+         - _is_token_json_temporally_valid
+         - valid_token_to_id
+
+        The first method checks the temporal validity of the payload
+        The second method detemines wether the token is invalid
+        """
+        payload_list = []
+
+        # Add a payload without 'iat' or 'exp' to the payload list
+        # to test we reject these (as we are choosing to)
```

Comment:
New sentence problem doesn't look very fixed and there's a few instances of this exact "sentence".

Reply:
ah yes, I probably just fixed the ones with a GitHub comment associated and forgot they would have been repeated